lib/ansible/modules/cloud/kubevirt/kubevirt_facts.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2018, KubeVirt Team <@kubevirt>
+# Apache License, Version 2.0
+# (see LICENSE or http://www.apache.org/licenses/LICENSE-2.0)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: kubevirt_facts
+
+short_description: Retrieve facts about a KubeVirt resources
+
+version_added: "2.5"
+
+author: KubeVirt Team (@kubevirt)
+
+description:
+  - Retrieve facts about a KubeVirt resources
+
+requirements:
+  - python >= 2.7
+  - kubevirt-python >= v0.6.0-177-gd265c3c
+'''
+
+EXAMPLES = '''
+# Gather facts about a VM called C(testvm) created on the namespace C(vms)
+- kubevirt_facts:
+    name: testvm
+    namespace: vms
+    kind: VirtualMachine
+  register: vm_facts
+- debug:
+    var: vm_facts
+
+# Gather facts for all VMIs in a cluster
+- kubevirt_facts:
+    kind: VirtualMachineInstance
+  register: all_vmis
+- debug:
+    var: all_vmis
+
+# Gather facts for all VMIRS in vms namespace
+- kubevirt_facts:
+    kind: VirtualMachineInstanceReplicaSet
+    namespace: vms
+  register: all_vms_vmirs
+- debug:
+    var: all_vms_vmirs
```

Comment:
I like this variable naming, it says what kind in what namespace. Notice earlier variable `vm_facts`, its name is kind of general and will be good if a user works with one VM only. Otherwise, I would change it for `testvm_facts` maybe. Just thinking loud, no need to correct it, it's just an example after all =)) 

PS, and in actual tests you used \<name\>_facts.